python/dolfinx/wrappers/fem.cpp

```diff
@@ -774,56 +774,52 @@ void declare_form(py::module& m, const std::string& type)
   std::string pyclass_name_form = std::string("Form_") + type;
   py::class_<dolfinx::fem::Form<T>, std::shared_ptr<dolfinx::fem::Form<T>>>(
       m, pyclass_name_form.c_str(), "Variational form object")
-      .def(
-          py::init(
-              [](const std::vector<std::shared_ptr<
-                     const dolfinx::fem::FunctionSpace>>& spaces,
-                 const std::map<
+      .def(py::init(
+               [](const std::vector<std::shared_ptr<
+                      const dolfinx::fem::FunctionSpace>>& spaces,
+                  const std::map<
```

Comment:
I think it would be better to use `py::dict` here, and `py::array_t`. `std::vector` doesn't cast efficiently for large arrays, and the `std::map` isn't passed directly to the C++ layer so the hidden cast from `py::dict` to `std::map` could be avoided.